pkg/cliplugins/bind/plugin/bind_compute.go

```diff
@@ -0,0 +1,295 @@
+		"APIExport to bind to this workspace for workload, each APIExoport should be in the format of <absolute_ref_to_workspace>:<apiexport>")
+	cmd.Flags().StringVar(&o.NamespaceSelector, "namespace-selector", o.NamespaceSelector, "Label select to select namespaces to create workload.")
+	cmd.Flags().StringSliceVar(&o.LocationSelectors, "location-selectors", o.LocationSelectors,
+		"A list of label selectors to select locations in the compute workspace to sync workload.")
+}
+
+// Complete ensures all dynamically populated fields are initialized.
+func (o *BindWorkloadOptions) Complete(args []string) error {
+	if err := o.Options.Complete(); err != nil {
+		return err
+	}
+
+	if len(args[0]) != 1 {
+		return fmt.Errorf("a compute workspace should be specified")
+	}
+
+	o.ComputeWorkspace = logicalcluster.New(args[0])
+
+	// if APIExport is not set use global kubernetes APIExpor and kubernetes APIExport in compute workspace
+	if len(o.APIExports) == 0 {
+		o.APIExports = []string{
+			"root:compute:kubernetes",
+			fmt.Sprintf("%s:kubernetes", o.ComputeWorkspace.String()),
+		}
+	}
+
+	// select all ns if namespace selector is not set
+	if len(o.NamespaceSelector) == 0 {
+		o.NamespaceSelector = labels.Everything().String()
+	}
+
+	// select all locations is location selectos is not set
+	if len(o.LocationSelectors) == 0 {
+		o.LocationSelectors = []string{labels.Everything().String()}
+	}
+
+	return nil
+}
+
+// Validate validates the BindOptions are complete and usable.
+func (o *BindWorkloadOptions) Validate() error {
+	return nil
+}
+
+// Run create a placement in the workspace linkind to the compute workspace
+func (o *BindWorkloadOptions) Run(ctx context.Context) error {
+	config, err := o.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+	userWorkspaceKcpClient, err := kcpclient.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	// build config to connect to compute workspace
+	computeWorkspaceConfig := rest.CopyConfig(config)
+	url, _, err := helpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return err
+	}
+
+	url.Path = path.Join(url.Path, o.ComputeWorkspace.Path())
+	computeWorkspaceConfig.Host = url.String()
+	computeWorkspaceKcpClient, err := kcpclient.NewForConfig(computeWorkspaceConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	err = o.hasSupportedSyncTargets(ctx, computeWorkspaceKcpClient)
+	if err != nil {
+		return err
+	}
+
+	err = o.applyPlacement(ctx, userWorkspaceKcpClient)
+	if err != nil {
+		return err
+	}
+
+	err = o.applyAPIBinding(ctx, userWorkspaceKcpClient)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func apiBindingName(clusterName logicalcluster.Name, apiExportName string) string {
+	hash := sha256.Sum224([]byte(clusterName.Path()))
+	base36hash := strings.ToLower(base36.EncodeBytes(hash[:]))
+	return fmt.Sprintf("%s-%s", apiExportName, base36hash[:8])
```

Comment:
Do we need to truncate `apiExportName` if it's longer than the maximum name length minus 8?

Reply:
@qiujian16 Do you agree it would be a good thing ?

Reply:
done